lib/models/User.js

```diff
@@ -8,18 +8,23 @@ var cypher = new Mcrypt('rijndael-128', 'ecb');
 var key = require('./../utils/get-mcrypt-key.js')();
 cypher.open(key);
 
+function decryptUsername(username) {
+    return cypher.decrypt(new Buffer(username.trim(), 'base64'))
+        .toString().replace(/\0/g, '');
+}
+
 module.exports = function(bookshelf) {
     return bookshelf.extend({
         tableName: 'cas_users',
+        idAttribute: 'username',
         initialize: function() {
-            this.on('fetched', this.decryptAttributes);
+            this.on('saved', this._utils.saveHist, this);
+            this.on('fetched', this.decryptAttributes, this);
         },
 
         decryptAttributes: function() {
-            this.set('username',
-                cypher.decrypt(new Buffer(this.get('username'), 'base64'))
-                    .toString().replace(/\0/g, '')
-            );
+            this.set('username', decryptUsername(this.get('username')));
+            delete this.attributes.password;
```

Comment:
We should never need to include the password column. This will hopefully prevent accidentally setting it to `null` when updating.